src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -1479,6 +1554,36 @@ def check_user_in_second_cluster(self, cluster_url, cluster_token, username,
             logger.info("User %s not found yet, retrying in 5 seconds", username)
             time.sleep(5)
 
+    def get_valid_realms_list(self):
+        valid_realms = []
+        realms_info = RgwMultisite().list_realms()
+        realm_list = realms_info.get('realms', [])
+        for realm_name in realm_list:  # pylint: disable=R1702
+            try:
+                realm_period = RgwMultisite().get_realm_period(realm_name)
+                master_zone_name = None
+                for zg in realm_period['period_map']['zonegroups']:
+                    if not zg.get('is_master'):
+                        continue
+                    for zone in zg.get('zones', []):
+                        if zone.get('id') == zg.get('master_zone'):
+                            if zone.get('endpoints'):
+                                master_zone_name = zone.get('name')
+                                break
+                if not master_zone_name:
+                    continue
+                zone_info = RgwMultisite().get_zone(master_zone_name)
+                system_key = zone_info.get('system_key', {})
+                access_key = system_key.get('access_key')
+                secret_key = system_key.get('secret_key')
+                if access_key and secret_key:
+                    valid_realms.append(realm_name)
```

Comment:
Does valid realm means ones having access and secret key and having master zone ? if you can elaborate more on what `valid realms` are here ? and how are they different from the normal realms ? 

It will be easier just to review further. Thanks

Reply:
valid realms are the ones that are eligible for replication. For a realm to be valid for replication it must have a master zone , valid endpoints and access/secret keys should be set.